create-event/src/App.js

```diff
@@ -0,0 +1,366 @@
+      headers: {
+        Authorization: `Bearer ${this.state.apiKey}`
+      }
+    })
+      .then(res => res.json())
+      .then(user =>
+        this.setState({ user: user.data }, () =>
+          this.snackbar(`Welcome ${user.data.attributes.first_name}!`)
+        )
+      )
+      .catch(err => this.snackbar(`'Unauthorized Access Token'`))
+  }
+
+  getEvent = () => {
+    fetch(`${host}/event/${this.state.event.id}?include=ticket_prices`)
+      .then(res => res.json())
+      .then(event =>
+        this.setState({
+          event: event.data,
+          tickets: event.included ? event.included : []
+        })
+      )
+      .catch(err => console.log(err))
+  }
+
+  updateEvent = message => {
+    const { event, tickets, deletedTickets, apiKey } = this.state
+
+    if (!this.validation()) {
+      return false
+    }
+
+    delete event.attributes.modified
+    delete event.attributes.created
+    delete event.attributes.live_on
+
+    const noEvent = isNaN(event.id)
+    if (noEvent) {
+      fetch(`${host}/event`, {
+        method: 'POST',
+        body: JSON.stringify({ data: event }),
+        headers: {
+          Authorization: `Bearer ${apiKey}`
+        }
+      })
+        .then(res => res.json())
+        .then(event => this.setState({ event: event.data }))
+        .catch(err => console.log(err))
+    } else {
+      // FIXME: Pass all event attributes in PATCH
+      fetch(`${host}/event/${event.id}`, {
+        method: 'PATCH',
+        body: JSON.stringify({
+          data: {
+            attributes: event.attributes,
+            id: event.id,
+            type: 'event'
+          }
+        }),
+        headers: { Authorization: `Bearer ${apiKey}` }
+      })
+        .then(res => res.json())
+        .then(event => this.setState({ event: event.data }))
+        .catch(err => console.log(err))
+    }
+
+    tickets.map((ticket, index) => {
+      delete ticket.attributes.modified
+      // Convert from paid to free ticket if price is $0
+      const freeTicket = ticket.attributes.price === 0
+      if (freeTicket) {
+        ticket.attributes.type = 'free'
+      }
+
+      const newTicket = isNaN(ticket.id)
+
+      if (newTicket) {
+        fetch(`${host}/ticket_price`, {
+          method: 'POST',
+          body: JSON.stringify({ data: ticket }),
+          headers: { Authorization: `Bearer ${apiKey}` }
+        })
+          .then(res => res.json())
+          .then(response => (tickets[index] = response.data))
+          .catch(err => console.log(err))
+      } else {
+        fetch(`${host}/ticket_price/${ticket.id}`, {
+          method: 'PATCH',
+          body: JSON.stringify({ data: ticket }),
+          headers: { Authorization: `Bearer ${apiKey}` }
+        })
+      }
+      return true
+    })
+
+    deletedTickets.map(ticket => {
+      fetch(`${host}/ticket_price/${ticket.id}`, {
+        method: 'DELETE',
+        headers: { Authorization: `Bearer ${apiKey}` }
+      })
+      return true
```

Comment:
need to remove tickets from this.state.deletedTickets once this fetch call succeeds, otherwise you will continue to make this call on updateEvent for tickets that have already been deleted